### DISCUSSION TITLE
Using AAD Group

### DISCUSSION THREAD

**Discussion by lune94** [Azure-Samples/ms-identity-python-webapp#74] [Q&A] [answered]
In a context of a web app I am using this library in order to authenticate my users via Azure Active Directory. I'd like to filter the access to pages according to AAD groups a user belongs to.
Example :
/index accessible only for members from AAD group 'project_users'
/admin accessible only for members from AAD groups 'project_contributor' and 'project_admin'

Does this library allow to do that ?

Regards

**Comment by rayluo** [comment]
@navyasric pointed out that we did not document this scenario in our docs, though a [Java sample did that](https://docs.microsoft.com/en-us/azure/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-active-directory#configure-and-compile-your-app). This Python web app sample does not currently provide such behavior out-of-box.

Perhaps you can look into the [claims of the id token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#claims-in-an-id_token) acquired by MSAL, probably with some [additional claims](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims), and then use its "groups" claim to decide whether your web app would serve/reject the request.

**Comment by rayluo** [comment]
@lune94 We do not currently have a Python web app sample for that, but this [".Net web app using groups" sample](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/master/5-WebApp-AuthZ/5-2-Groups/README.md) and [this video "Using Security Groups and Application Roles in your apps"](https://www.youtube.com/watch?v=V8VUPixLSiM) will give you a good understanding on the details.

**Comment by RyanPicco** [comment, marked as answer]
@lune94 Though I'm not using groups in this exact way, you can define app roles in the [manifest](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps#:~:text=In%20the%20Azure%20Active%20Directory,adding%20all%20your%20Application%20Roles.)

So after defining the roles you can assign roles to your users (or in your case groups) which are allowed access, then their role will be returned in id_token_claims as "roles", from there you should be able to gate access based on their role value being in a list of acceptable roles, you could write this into a decorator /w args for easy application to each endpoint.
